### ISSUE TITLE
Bug: False positive in no-implied-eval when using non-global execScript function

### ISSUE THREAD

**Issue by kjots** [eslint/eslint#19923]
### Environment

Node version: v22.17.0
npm version: v10.9.2
Local ESLint version: v9.30.1 (Currently used)
Global ESLint version: Not found
Operating System: darwin 24.5.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [
  {
    rules: {
      'no-implied-eval': 'error'
    }
  }
];
```
</details>

```js
function execScript(string) {
  console.log("This is not your grandparent's execScript().");
}

execScript('wibble');
```

### What did you expect to happen?

No error should occur.

### What actually happened?

```
…/index.js
  5:1  error  Implied eval. Consider passing a function instead of a string  no-implied-eval

✖ 1 problem (1 error, 0 warnings)
```

### Link to Minimal Reproducible Example

https://github.com/kjots/eslint-issue-no-implied-eval-exec-script

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

The [`no-implied-eval`](https://eslint.org/docs/latest/rules/no-implied-eval) rule will trigger a false positive for `execScript()` even when the code under test does not make use of the global `execScript()` function.

**Comment by snitin315** [comment]
Looks like a bug to me, working on a fix.

[Playground Link](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8taW1wbGllZC1ldmFsOiBcImVycm9yXCIqL1xuXG5mdW5jdGlvbiBzZXRUaW1lb3V0KGZvbykge1xuICByZXR1cm4gZm9vO1xufVxuXG5zZXRUaW1lb3V0KCd0ZXN0Jyk7Iiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19)

**Comment by TKDev7** [comment]
Do we not consider this case an edge case? If so, I believe similar cases exist in other rules.

**Comment by kjots** [comment]
> Do we not consider this case an edge case? If so, I believe similar cases exist in other rules.

I would imagine in 2025 that use of the global `execScript()` function in Internet Explorer is the edge case, I don't think users would expect to not be able to use the `execScript` name due to the presence of a [deprecated function](https://web.archive.org/web/20140203084900/https://msdn.microsoft.com/en-us/library/ms536420(VS.85).aspx) in an obsolete browser (I certainly didn't).

Another issue with using the `no-implied-eval` rule to capture use of the `execScript()` function is that the error message is wrong - it suggests "passing a function instead of a string", however `execScript()` does not accept a function so that advice cannot be followed. The only resolution would be to _not_ use `execScript()` at all, which in fact is also stated in the [rule docs](https://eslint.org/docs/latest/rules/no-implied-eval#:~:text=and%20avoid%20execScript()).

My opinion is that checking for `execScript()` should be removed from the `no-implied-eval` rule altogether, and what I'm sure by now is the very small number of people that still need to capture use of the global `execScript()` function could instead use [`no-restricted-globals`](https://eslint.org/docs/latest/rules/no-restricted-globals) which would seem to be the more appropriate rule for this use case.

**Comment by TKDev7** [comment]
@kjots I fully agree with your perspective on this. I've opened #19927 to formally propose removing execScript detection from no-implied-eval, where we can discuss this change with the team.

**Comment by nzakas** [comment]
@snitin315 don't forget to move the issue in the triage board.

I agree that this is a bug and we should fix it. We should only be checking global references for `execScript()`.

I disagree with removing the check altogether. We know that people use ESLint to support all kinds of environments, including those down to ES3. This check likely still has value for some users and once this bug is fixed, I can't think of a good argument for removing something that is working correctly.

**Comment by snitin315** [comment]
I agree with @nzakas on both the points, we should just check if the function is global reference or not.
